lib/ecstatic/status-handlers.js

```diff
@@ -2,58 +2,58 @@
 
 const he = require('he');
 
+function wrap(statusCode, fn) {
+  return (res, next, opts) => {
+    if (typeof next === 'function') {
```

Comment:
This is inconsistent with the current behavior, right?

By my research, it looks like current behavior is to only check handleError in 404 cases and that it's not wired up literally anywhere else, and that prior to this change "next" was ignored, even though it was being passed in. 🤔 Not sure what the intent was there.

So there are a couple of things that would be good to see, if you're game:

- [ ] A documentation update to make this behavior more clear - shouldn't be a big deal, just describing that handleError: false will fall-through for whatever set of status codes
- [ ] Some tests that check that next is called properly depending on the state of handleError. I realize that the tests are kind of a shitshow. If this is a relatively low lift 🤞 great! If not, maybe not a big deal.

Reply:
Rather than making things complicated... I think I've fixed it so that it is consistent with the current behavior.

Reply:
I have to admit I don't understand the current behavior... but let's fix the current behavior first and then consider changing it...

Reply:
Honestly, I don't either. This is a project I wrote a long time ago and I haven't held the whole thing in my head for a while. Honestly there are a lot of things I would do different in a rewrite, but the test harness is awkward enough that I think a refactor that size would be challenging.

Anyway, ignoring next(), or removing its use from the call sites is fine by me, though one of us should flag this as an issue.

Reply:
Written up here: https://github.com/jfhbrook/node-ecstatic/issues/251 feel free to add notes if you have any.